packages/clerk-js/src/core/fapiClient.ts

```diff
@@ -204,9 +204,10 @@ export default function createFapiClient(clerkInstance: Omit<Clerk, 'proxyUrl'>)
     };
 
     const json: FapiResponseJSON<T> = await response.json();
-    const fapiResponse: FapiResponse<T> = Object.assign(response, {
-      payload: { ...json, meta: { ...json?.meta, responseHeaders } },
-    });
+    const fapiResponse: FapiResponse<T> = Object.assign(response, { payload: json });
+    if (fapiResponse.payload) {
+      fapiResponse.payload.meta = { ...json?.meta, responseHeaders };
+    }
```

Comment:
The issue is related to `...json` which causes an `array` to be converted to `object`. Since the `SessionWithActivities#retrieve()` an array response is being expected, the above code was converting the fapiResponse to an object which causes the code to fail because a `.map` is not a function of object.

Reply:
❓ Was this changed recently?

Reply:
yes `3 days ago`